src/internal/connector/discovery/api/users.go

```diff
@@ -220,3 +326,31 @@ func validateUser(item any) (models.Userable, error) {
 
 	return m, nil
 }
+
+func convertInterfaceToString(ctx context.Context, data interface{}, userInfo *UserInfo) string {
+	var (
+		ok          bool
+		dataPointer *string
+		value       string
+	)
+
+	dataPointer, ok = data.(*string)
+	if !ok {
+		logger.Ctx(ctx).Infof("error getting mailboxSettings")
+
+		userInfo.ErrGetMailBoxSetting = "not found"
```

Comment:
Mutating the userInfo is a side effect that shouldn't be part of this function.  You could instead catch an empty string when handling the result for the same consequence, and no side effect.

Alternatively, pass back an error sentinel, like so:
```go
var ErrMailBoxSettingsNotFound = clues.New("mailbox settings not found")

s, err := convert(...)
if err != nil {
  userInfo.ErrGetMailboxSetting = err
}

func convert(...) (string, error) {
  if !ok {
    return "", ErrMailBoxSettingsNotFound
  }
} 
```